codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -773,30 +507,40 @@ Gets the earliest queueItem from thgetNodeEarliestEventWithUpdatedQueue.
 Finds the earliest queue item across all registers and then returns that
 queue item with an update fetch state.
 */
-let getEarliestEvent = (self: t) => {
-  let earliestItemInRegisters = {
-    let registerWithEarliestQItem = self.baseRegister->findRegisterIdWithEarliestQueueItem
-    //Can safely unwrap here since the id is returned from self and so is guarenteed to exist
-    self.baseRegister->popQItemAtRegisterId(~id=registerWithEarliestQItem)->Utils.unwrapResultExn
+let getEarliestEvent = (fetchState: t) => {
+  let earliestItemInRegisters = switch fetchState.registers {
+  | [r] => r->getEarliestEventInRegister
+  | registers => {
+      let item = ref(registers->Js.Array2.unsafe_get(0)->getEarliestEventInRegister)
+      for idx in 1 to registers->Js.Array2.length - 1 {
+        let register = registers->Js.Array2.unsafe_get(idx)
+        let registerItem = register->getEarliestEventInRegister
+        if registerItem->qItemLt(item.contents) {
+          item := registerItem
+        }
+      }
+      item.contents
+    }
```

Comment:
Is this much faster than reduce?

If so looks like we should just make this into our own reduce util?

Reply:
I suppose the speed difference with our number of registers is neglectable, but I think regarding the indexer code we should strive for the better, especially when it doesn't cost anything.

I'm not really into the reduce util. There's only one pro in using reduce - readability, but this is a very arguable and I find the procedure-style code equally readable, especially when it's scoped inside of a small function covered with tests.

Reply:
Well there's a number of places this pattern is used. If it's faster we should make a util for it try convert all other reduce use cases to this IMO.

It's not really readability since as you say procedural code often reads easier, rather composability and safety from the off by ones/bad index accessing etc. Those bugs can creep in from time to time.